.github/workflows/promote-docker-image.yml

```diff
@@ -0,0 +1,39 @@
+name: Promote docker image to latest or stable
+on: 
+  workflow_dispatch:
+    inputs:
+      tag:
+        type: string
+        description: "e.g: v2.8.4"
+        required: true
+      promote-to:
+        type: choice
+        description: "Promote to latest or stable"
+        options:
+          - "latest"
+          - "stable"
+        required: true
+env:
+  REGISTRY: "docker.io"
+  IMAGE: ${{ github.repository }}
+  TAG: ${{ github.event.inputs.tag }}
+  NEW_TAG: ${{ github.event.inputs.promote-to }}
+jobs:
+  promote:
+    runs-on: ubuntu-latest
+    container:
+      image: quay.io/skopeo/stable:v1.15
+    permissions:
+      contents: read
+      id-token: write
+    steps:
+      - name: Load Secrets from Vault
+        uses: rancher-eio/read-vault-secrets@main
+        with:
+          secrets: |
+            secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials username | DOCKER_USERNAME ;
+            secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials password | DOCKER_PASSWORD
+      - name: Skopeo login
+        run: echo ${{ env.DOCKER_PASSWORD }} | skopeo login ${{ env.REGISTRY }} --username ${{ env.DOCKER_USERNAME }} --password-stdin
```

Comment:
LGTM

Just one question, have we confirmed that credentials will not leaked in GHA logs here?

Reply:
@macedogm Looking at other runs, I'm pretty confident they will not be leaked: https://github.com/rancher/rancher/actions/runs/10188357510/job/28192473030#step:4:2